pallets/msa/src/lib.rs

```diff
@@ -1913,23 +2136,26 @@ impl<T: Config> Pallet<T> {
 	/// # Errors
 	/// * [`Error::DuplicateProviderRegistryEntry`] - a ProviderRegistryEntry associated with the given MSA id already exists.
 	///
-	pub fn create_provider_for(
+	pub fn upsert_provider_for(
 		provider_msa_id: MessageSourceId,
 		payload: ProviderRegistryEntry<
 			T::MaxProviderNameSize,
 			T::MaxLanguageCodeSize,
 			T::MaxLogoCidSize,
 			T::MaxLocaleCount,
 		>,
+		is_update: bool,
 	) -> DispatchResult {
 		Self::update_logo_storage(&payload)?;
 		ProviderToRegistryEntry::<T>::try_mutate(
 			ProviderId(provider_msa_id),
 			|maybe_metadata| -> DispatchResult {
-				ensure!(
-					maybe_metadata.take().is_none(),
-					Error::<T>::DuplicateProviderRegistryEntry
-				);
+				if !is_update {
```

Comment:
there is an issue with `update_logo_storage` . it doesn't look like we ever remove the old ones when upserting.